src/components/Chat/Chat.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { PropsWithChildren } from 'react';
+
+import { useChat } from './hooks/useChat';
+
+import { ChatProvider } from '../../context/ChatContext';
+import { TranslationProvider } from '../../context/TranslationContext';
+
+import type { StreamChat } from 'stream-chat';
+import type { Streami18n } from '../../i18n/Streami18n';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type ChatProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  /** The StreamChat client object */
+  client: StreamChat<At, Ch, Co, Ev, Me, Re, Us>;
+  /** Instance of Stream i18n */
+  i18nInstance?: Streami18n;
+  /** Initial status of mobile navigation */
+  initialNavOpen?: boolean;
+  /**
+   *
+   * Theme could be used for custom styling of the components.
+   *
+   * You can override the classes used in our components under parent theme class.
+   *
+   * e.g. If you want to build a theme where background of message is black
+   *
+   * ```
+   *  <Chat client={client} theme={demo}>
+   *    <Channel>
+   *      <MessageList />
+   *    </Channel>
+   *  </Chat>
+   * ```
+   *
+   * ```scss
+   *  .demo.str-chat {
+   *    .str-chat__message-simple {
+   *      &-text-inner {
+   *        background-color: black;
+   *      }
+   *    }
+   *  }
+   * ```
+   *
+   * Built in available themes:
+   *
+   *  - `messaging light`
+   *  - `messaging dark`
+   *  - `team light`
+   *  - `team dark`
+   *  - `commerce light`
+   *  - `commerce dark`
+   *  - `livestream light`
+   *  - `livestream dark`
+   */
+  theme?: string;
```

Comment:
should probably create a `Theme` type for this unless there's the potential for it to include anything more than this from our clients own customizations